src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -316,14 +317,31 @@ def _find_node_to_breakdown(
                         # don't traverse subtrees if parent is a valid candidate
                         next_level.append(child)
 
+                    relaxed_validity_status, _ = self._is_node_valid_to_breakdown(
```

Comment:
if the node is already a valid candidate, seems we do not need to check whether it is relaxed or not anymore, right?

Reply:
yeah. that is true.

Reply:
updated now